src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java

```diff
@@ -0,0 +1,397 @@
+
+		List<AuthorizationIntraCloudRequestDTO> dtoList = new ArrayList<>();
+		Long consumerId = null;
+		List<Long> providerId;
+		List<Long> serviceDefinitionId;
+		List<Long> interfaceIds;
+
+		for (int i = 0; i < rules.size(); i++) {
+
+			providerId = new ArrayList<>();
+			serviceDefinitionId = new ArrayList<>();
+			interfaceIds = new ArrayList<>();
+
+			try {
+				consumerId = getSystemIdByInfo(rules.get(i).getConsumer());
+
+				providerId.add(getSystemIdByInfo(rules.get(i).getProvider()));
+
+				serviceDefinitionId.add(serviceDefinitionToId(rules.get(i).getService()));
+
+				for (String interfaceName : rules.get(i).getInterfaces()) {
+					interfaceIds.add(interfaceNameToId(interfaceName));
+
+				}
+
+				dtoList.add(new AuthorizationIntraCloudRequestDTO(consumerId, providerId, serviceDefinitionId,
+						interfaceIds));
+			} catch (Exception e) {
+				logger.error("Could not create authorization rule: " + rules.get(i).toString() + ", reason: "
+						+ e.getMessage());
+			}
+		}
+		return dtoList;
+
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private void addSingleRule(final AuthorizationIntraCloudRequestDTO ruleToAdd) {
+		Map<String, String> authorizationUri = null;
+		try {
+			authorizationUri = getAuthorizationUri();
+		} catch (Exception e) {
+			logger.error(e.getMessage());
+		}
+
+		logger.info("Sending the POST request for the following authorization rule: " + ruleToAdd.toString());
+		AuthorizationIntraCloudListResponseDTO response = arrowheadService.consumeServiceHTTP(
+				AuthorizationIntraCloudListResponseDTO.class, HttpMethod.POST,
+				Utilities.createURI(authorizationUri.get("scheme"), authorizationUri.get("host"),
+						Integer.parseInt(authorizationUri.get("port")),
+						authorizationUri.get("path") + ConsumerConstants.QUERY_AUTH_INTRA_CLOUD),
+				null, ruleToAdd);
+		if (response == null) {
+			logger.error("Could not apply the following authorization rule: " + ruleToAdd.toString());
+		} else {
+			logger.info("Successfully applied rule with id: " + response.getData().getFirst().getId());
+		}
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private List<Integer> getSystemIdsToDelete(final List<AuthRule> rules) {
+
+		List<Integer> result = new ArrayList<>();
+
+		for (AuthRule rule : rules) {
+			try {
+				int id = (int) getSystemIdByInfo(rule.getConsumer());
+				if (!result.contains(id)) {
+					result.add(id);
+				}
+			} catch (Exception e) {
+				logger.error(e);
+			}
+
+		}
+
+		return result;
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	// the system string can be systemname or metadata
+	// (in case it is metadata, it must contain '=')
+	private long getSystemIdByInfo(final String systemInfo) throws Exception {
```

Comment:
This method is wrong. In case of metadata, it only returns the first system that matches that metadata. It should return all.